nuxeo-core/nuxeo-core-schema/src/main/java/org/nuxeo/ecm/core/schema/DocumentTypeImpl.java

```diff
@@ -39,22 +39,40 @@
 
     protected PrefetchInfo prefetchInfo;
 
+    protected Set<String> subtypes;
+
+    protected Set<String> blacklistedSubtypes;
+
+    protected Set<String> allowedSubtypes;
+
     /**
      * Constructs a document type. Schemas and facets must include those from the super type.
      */
     public DocumentTypeImpl(String name, DocumentType superType, List<Schema> schemas, Collection<String> facets,
-            PrefetchInfo prefetchInfo) {
+            PrefetchInfo prefetchInfo, Collection<String> subtypes, Collection<String> blacklistedSubtypes) {
```

Comment:
As it's a public API, you should keep the old method there and add a new one with your new args.